src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -253,12 +253,20 @@ void GcsNodeManager::OnNodeFailure(const NodeID &node_id) {
 }
 
 void GcsNodeManager::Initialize(const GcsInitData &gcs_init_data) {
-  for (const auto &item : gcs_init_data.Nodes()) {
-    if (item.second.state() == rpc::GcsNodeInfo::ALIVE) {
-      AddNode(std::make_shared<rpc::GcsNodeInfo>(item.second));
-    } else if (item.second.state() == rpc::GcsNodeInfo::DEAD) {
-      dead_nodes_.emplace(item.first, std::make_shared<rpc::GcsNodeInfo>(item.second));
-      sorted_dead_node_list_.emplace_back(item.first, item.second.timestamp());
+  for (const auto &[node_id, node_info] : gcs_init_data.Nodes()) {
```

Comment:
could there be a case where the gcs_init_data doesn't have a full list of nodes so we fail to notify some nodes?

Reply:
That means the node will not be in the cluster. So no need to ask for resubscribe. If the node register later, it'll do resubscribe there.

Reply:
It will be nice to leave a comment to convince people that this scheme is correct. e.g. is Raylet guaranteed to register with GCS before subscribing to GCS?